enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/CORS-2840
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Enabling Cluster-API-based Installations via openshift-install
+
+## Summary
+
+This enhancement discusses how `openshift-install` can use
+cluster-api (CAPI) providers to provision cluster infrastructure,
+without requiring access to an external management cluster or a local container runtime.
+By running a Kubernetes control plane and CAPI-provider controllers as
+subprocesses on the installer host, `openshift-install` can use CAPI and its
+providers in a similar manner to how Terraform and its providers are currently
+being used.
+
+## Motivation
+
+1. OpenShift Alignment with CAPI: CAPI offers numerous potential benefits;
+such as: day-2 infrastructure management, an API for users to edit cluster
+infrastructure, and upstream collaboration. Installer support for CAPI would
+be foudational for adopting these benefits.
+
+2. Terraform BSL License Change: due to the restrictive license change of
+Terraform, `openshift-install` needs a framework to replace the primary
+tool it used to provision cluster infrastructure. In addition to the benefits
+listed above, CAPI provides solutions for the biggest gaps left by Terraform:
+a common API for specifying infrastructure across cloud providers and robust
+infrastructure error handling.
+
+3. Streamline Installer development: a common pattern for Installer development
+has been to reimplement control-plane features in Terraform that have already been
+delivered upstream. By utilizing CAPI providers for Day-0 provisioning, our development
+practices will remove this duplication and become more efficient.
+
+### User Stories
+
+- As an existing user/client of the installer, I want backwards compatibility so that I can continue to use the installer (e.g. `create cluster`) in the same manner and with existing automation.
+- As a security analyst, I want the installer image to be free of Terraform and related dependencies to decrease surface area for vulnerabilities.
+- As an advanced user or cluster administrator, I want to be able to edit the CAPI infrastructure manifests so that I can customize control-plane infrastructure.
+
+
+
+### Goals
+
+- To provide a common user and developer experience when installing and developing across cloud platforms
+- To be backwards compatible and fully satisfy the requirements of install-config type APIs.
+- To keep the user experience for day-zero operations unchanged or improved.
+- To not require any new runtime dependencies.
+- To provide an extensible framework to plug-in new infrastructure cloud providers.
+- To maintain compatibility with hive, particularly regarding `destroy` functionality
```

Comment:
Log collection came up as a potential challenge for hive. The "log bundle" cited elsewhere wouldn't have an obvious place to be stored. And it's possible there are other aspects of this proposal that could impact hive. It may be worth figuring out in advance what the total impact is, and then whether it makes sense to worry about, or whether we won't expect hive to have to deal with the changes.

Reply:
My understanding is we need to continue supporting Hive, for now, so any gaps need to be identified and we need to know how they'll be closed. Discussions about more significant changes, including design changes or potential deprecation, would come after this work is done.

Reply:
For the purpose of this proposal, which is primarily to remove the use of Terraform from our installer codebase; we're focusing on the work outside of HIVE. I'll add a point to the future work section to revisit hive once day 2 work starts